### ISSUE TITLE
[BUG] fatal error in SQANTI3_report.R 

### ISSUE THREAD

**Issue by diekhans** [ConesaLab/SQANTI3#278]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you loaded the SQANTI3.env conda environment?

- [ ] I have loaded the SQANTI3.env conda environment

### Problem description

SQANTI3_report.R fails with. cryptic error:
```
Error in `[.data.frame`(t3.a.SJ, , c(1, 5, 6)) :
  undefined columns selected
Calls: rbind ... standardGeneric -> eval -> eval -> eval -> [ -> [.data.frame
Execution halted
```

### Code sample

data files and script to reproduce the problem are here

https://hgwdev.gi.ucsc.edu/~markd/crg/crg-sqanti-err.tar.gz

data and script to do the full squant3 run is here

https://hgwdev.gi.ucsc.edu/~markd/crg/crg-sqanti-err-full.tar.gz

### Error

Error in `[.data.frame`(t3.a.SJ, , c(1, 5, 6)) :
  undefined columns selected
Calls: rbind ... standardGeneric -> eval -> eval -> eval -> [ -> [.data.frame
Execution halted

**Comment by alexpan00** [comment]
Hi,

I think I have suffered this error myself before, it was caused by not having any transcript with non-canonical junctions. This also happens in the classification file you have provided.

Thanks for reporting this bug,
Alejandro.

**Comment by TianYuan-Liu** [comment]
> Hi,
> 
> I think I have suffered this error myself before, it was caused by not having any transcript with non-canonical junctions. This also happens in the classification file you have provided.
> 
> Thanks for reporting this bug, Alejandro.

Hi Alejandro, thanks for reporting this bug and sharing your experience. I'll address this issue related to transcripts with non-canonical junctions in the R report to prevent future occurrences. Your input is invaluable in improving our processes.

**Comment by carolinamonzo** [comment]
This error has now been fixed in v5.2.1. Thank you Mark and Tian!
